src/Paprika/Store/PagedDb.cs

```diff
@@ -370,14 +373,28 @@ public void Dispose()
             db.ReportReads(Volatile.Read(ref _reads));
             _disposed = true;
             db.DisposeReadOnlyBatch(this);
+
+            ReturnCacheToPool();
+
+            void ReturnCacheToPool()
+            {
+                var idCache = _idCache;
+                _idCache = null!;
+                ref var cache = ref s_cache;
```

Comment:
@benaadams Should `s_cache` access be `volatile`? No barrier on return.

Reply:
Is threadstatic so doesn't matter too much; I take a `ref` of `s_cache` because accessing (finding) a threadstatic is more expensive than regular field, so once have its memory location work with that.

I use `Interlocked.Exchange` to get; not because of thead worry issues, but is single fast instruction whereas it would be multi and messier with taking a `ref` (for this reason) geting value, conditionally blanking  etc